process/block/shardblock.go

```diff
@@ -116,6 +120,16 @@ func NewShardProcessor(arguments ArgShardProcessor) (*shardProcessor, error) {
 	}
 	metaBlockPool.RegisterHandler(sp.receivedMetaBlock)
 
+	miniBlockPool := sp.dataPool.MiniBlocks()
+	if miniBlockPool == nil {
+		return nil, process.ErrNilMiniBlockPool
+	}
+	log.Debug("registering miniblock pool handler")
+	miniBlockPool.RegisterHandler(sp.receivedMiniBlock)
+
+	sp.startOfEpochData = map[string]startOfEpochData{}
```

Comment:
move all the data gathering to the validator statistics pre processor.

Reply:
fixed